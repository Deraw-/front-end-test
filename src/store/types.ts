// tslint:disable-next-line:no-empty-interface
export interface RootState {}

export interface Product extends Object {
	id: number;
	quantity: number;
}

export interface CartProduct extends Product {
	name: string;
	price: number;
	image: string;
}

// This is used to properly type components using mappers from Vuex
// see https://github.com/vuejs/vuex/pull/1121
// tslint:disable-next-line: no-empty-interface
export default interface VuexBindings {
	cartProducts: CartProduct[];
}
