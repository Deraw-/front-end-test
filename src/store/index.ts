import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import VuexPersistence from 'vuex-persist';
import { RootState } from './types';

// Modules
import cart from './modules/cart';
import products from './modules/products';

// The store is saved in the browser's session
const vuexLocal = new VuexPersistence({
	storage: window.sessionStorage
});

Vue.use(Vuex);

// See https://vuex.vuejs.org/fr/getting-started.html for help
const store: StoreOptions<RootState> = {
	strict: true,
	state: {},
	// See https://vuex.vuejs.org/guide/modules.html for more info on modules
	modules: {
		cart,
		products
	},
	plugins: [
		vuexLocal.plugin
	]
};

export default new Vuex.Store<RootState>(store);
