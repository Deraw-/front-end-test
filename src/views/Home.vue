<template>
	<div class="home-ctn">
		<section class="banner">
			<h1 class="main-title">
				PÃ¢tisserie
			</h1>

			<img
				src="/img/background.jpg"
				alt=""
			>
		</section>

		<section class="page">
			<ProductList :products="all" />
		</section>
	</div>
</template>

<script lang="ts">
	import Vue from 'vue';
	import { mapState, mapActions } from 'vuex';

	import ProductList from '@/components/ProductList.vue';

	export default Vue.extend({
		name: 'Home',
		components: {
			ProductList
		},
		computed: {
			...mapState('products', ['all'])
		},
		mounted() {
			this.setAllProducts();
		},
		methods: {
			...mapActions('products', ['setAllProducts'])
		}
	});
</script>

<style lang="scss" scoped>
	.home-ctn {
		width: 100%;
	}

	.banner {
		overflow: hidden;
		position: relative;

		display: flex;
		flex: 1 0 auto;
		align-items: center;
		justify-content: center;

		min-height: 350px;
		height: calc(100vh - 64px - 185px); // viewport - toolbar - gap to see content

		img {
			width: 100%;
			height: 100%;
			object-fit: cover;
		}
	}

	.main-title {
		color: #fff;
		z-index: 1;
		position: absolute;

		margin-top: -25px;

		line-height: 1;
		font-weight: 400;
		font-size: calc(4vw + 2rem);
		font-family: serif;

		text-align: center;
		text-shadow: 0 0 100px black, 0 0 100px black;

		&::after {
			left: 50%;
			width: 50%;
			content: '';
			height: 5px;
			bottom: -1vh;
			background: #fff;
			position: absolute;
			transform: translateX(-50%);
		}
	}

	@media only screen and (max-width: 600px) {
		.banner {
			height: calc(100vh - 64px - 155px);
		}
	}
</style>
