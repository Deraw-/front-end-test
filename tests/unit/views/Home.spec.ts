import { shallowMount } from '@vue/test-utils';
import Home from '@/views/Home.vue';

import localVue from 'tests-unit/index';
import store from '@/store/';

// Build component before each test
let wrapper: any;
beforeEach(() => {
	wrapper = shallowMount(Home, {
		localVue,
		store
	});
});

// Tests
describe('Home.vue', () => {
	it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
