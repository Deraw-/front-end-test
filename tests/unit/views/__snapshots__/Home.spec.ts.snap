// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders correctly 1`] = `
<div class="home-ctn">
  <section class="banner">
    <h2 class="main-title">
      views.home.title
    </h2> <img src="/img/background.jpg" alt="">
  </section>
  <section class="page">
    <productlist-stub products=""></productlist-stub>
  </section>
</div>
`;
